modules.d/01fips/fips-boot.sh

```diff
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-if ! fipsmode=$(getarg fips) || [ $fipsmode = "0" ]; then
+if ! fipsmode=$(getarg fips) || [ "$fipsmode" = "0" ]; then
```

Comment:
This looks like it really should be using `getargbool 0 fips`, i.e. something like:

```
if ! getargbool 0 fips; then
```
See https://github.com/dracutdevs/dracut/blob/master/modules.d/99base/dracut-lib.sh#L245

(and for cherries on top maybe invert logic and swap if/else block to make reading easier)

Reply:
I'm not sure this will really work. 
If you don't have fips on command line at all. It will behave the same way as if you have `fips=0` which is not what you want. That is the reason why the original conditional is so complicated AFAIK.

What you want here is to run the first condition only if you have `fips=0` the second in case `fips` and the third when there is `fips=1`. This is not easily doable by using `getargbool` thanks to the default value.

Reply:
I don't quite follow.  the general behavior would be the same, wouldn't it, as it only depends on those two checks that would decided the same in all cases (no param, `fips`, `fips=0` and `fips=1`)?

There is only these two `getargs` that check for the `fips` parameter. `DRACUT_FIPS_MODE` depends on fips module being included at build time.

Reply:
OK, after some more thinking it should behave the same as the original code.. but it doesn't.

After your proposed change missing `fips`, `fips=0`, `fips` and `fips=1` behaves the same, they all break bootup process. Unfortunately, I really don't know why, it just froze because `install.img` can't be downloaded by curl.

I guess I have to do more testing. The bad thing is, that I'm not able to get into console because it just freeze :(. And it looks like it behaves correct in the rd.break...